Disable autocorrect in server settings input fields

bernhard-fp opened the pull request:
Set textfield inputtypes to textNoSuggestions

First pull-request.
I hope I have understood the ticket correctly.

Fixes #7361

cketti commented:
Welcome @bernhard-fp :wave:

Thanks for wanting to contribute.

We're currently rewriting the account setup using Jetpack Compose. The new code can be found here: 
- https://github.com/thunderbird/thunderbird-android/blob/main/feature/account/server/settings/src/main/kotlin/app/k9mail/feature/account/server/settings/ui/incoming/IncomingServerSettingsScreen.kt
- https://github.com/thunderbird/thunderbird-android/blob/main/feature/account/server/settings/src/main/kotlin/app/k9mail/feature/account/server/settings/ui/outgoing/OutgoingServerSettingsScreen.kt

The layout you've changed will be removed soon. So sadly this is not what we're looking for.

Issue #7361 contains a comment with links to the relevant documentation for Compose. Since we're wrapping Material composables to create our own design system, support for being able to specify `KeyboardOptions` has to be added in the module `:core:ui:compose:designsystem` first, e.g. [`TextFieldOutlined`](https://github.com/thunderbird/thunderbird-android/blob/main/core/ui/compose/designsystem/src/main/kotlin/app/k9mail/core/ui/compose/designsystem/atom/textfield/TextFieldOutlined.kt). Then the screens for incoming and outgoing server settings can be changed to make use of that new feature.

bernhard-fp commented:
@cketti Thank you. So I think I can start adding a new value keyboardOptions to IncomingFormItems->TextInput and set the autoCorrect value to false in the both screens you mentioned.
Per default it would be KeyboardOptions()

cketti commented:
You should be able to copy the approach from one of the Material Composables we're wrapping, e.g.
https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/material/material/src/commonMain/kotlin/androidx/compose/material/OutlinedTextField.kt;l=149;drc=0ea8ee615a4f669d7ecfbcb657704a937fb5129d

Using `KeyboardOptions.Default` will avoid having to create a new instance when you don't want to override the default behavior.

bernhard-fp commented:
Hi @cketti, I reverted this first commit and added the KeyboardOptions to the belonging classes.
Unfortunately I can not test because on my Pixel7 and on my emulator I do not see any auto corrections.
But now autoCorrect=false is explicitly set.
Thanks for reviewing :-)

cketti commented on feature/account/server/settings/src/main/kotlin/app/k9mail/feature/account/server/settings/ui/incoming/content/IncomingFormItems.kt:
```diff
@@ -53,6 +54,9 @@ internal fun LazyListScope.incomingFormItems(
             onTextChange = { onEvent(Event.ServerChanged(it)) },
             label = stringResource(id = R.string.account_server_settings_server_label),
             contentPadding = defaultItemPadding(),
+            keyboardOptions = KeyboardOptions.Default.copy(
+                autoCorrect = false,
+            ),
```
`KeyboardOptions` has default values for all parameters. So writing this is equivalent and shorter:

```suggestion
            keyboardOptions = KeyboardOptions(autoCorrect = false),
```

bernhard-fp replied:
Yes, right. I also added it in OutgoingFormItems.kt

bernhard-fp commented:
@cketti Could you please test if it matches the desired functionality?

cketti commented on feature/account/server/settings/src/main/kotlin/app/k9mail/feature/account/server/settings/ui/incoming/content/IncomingFormItems.kt:
```diff
@@ -53,6 +54,9 @@ internal fun LazyListScope.incomingFormItems(
             onTextChange = { onEvent(Event.ServerChanged(it)) },
             label = stringResource(id = R.string.account_server_settings_server_label),
             contentPadding = defaultItemPadding(),
+            keyboardOptions = KeyboardOptions(
+                autoCorrect = false,
+            ),
```
There's no need for this to span three lines.

```suggestion
            keyboardOptions = KeyboardOptions(autoCorrect = false),
```

bernhard-fp replied:
Fixed

cketti commented on feature/account/server/settings/src/main/kotlin/app/k9mail/feature/account/server/settings/ui/incoming/content/IncomingFormItems.kt:
```diff
@@ -95,6 +99,9 @@ internal fun LazyListScope.incomingFormItems(
             onTextChange = { onEvent(Event.UsernameChanged(it)) },
             label = stringResource(id = R.string.account_server_settings_username_label),
             contentPadding = defaultItemPadding(),
+            keyboardOptions = KeyboardOptions.Default.copy(
+                autoCorrect = false,
+            ),
```
This can also be changed to the single line constructor invocation.

cketti commented on feature/account/server/settings/src/main/kotlin/app/k9mail/feature/account/server/settings/ui/outgoing/content/OutgoingFormItems.kt:
```diff
@@ -42,6 +43,9 @@ internal fun LazyListScope.outgoingFormItems(
             label = stringResource(id = R.string.account_server_settings_server_label),
             isRequired = true,
             contentPadding = defaultItemPadding(),
+            keyboardOptions = KeyboardOptions(
+                autoCorrect = false,
+            ),
```
This should also be a single line.

cketti commented on feature/account/server/settings/src/main/kotlin/app/k9mail/feature/account/server/settings/ui/outgoing/content/OutgoingFormItems.kt:
```diff
@@ -87,6 +91,9 @@ internal fun LazyListScope.outgoingFormItems(
                 label = stringResource(id = R.string.account_server_settings_username_label),
                 isRequired = true,
                 contentPadding = defaultItemPadding(),
+                keyboardOptions = KeyboardOptions.Default.copy(
+                    autoCorrect = false,
+                ),
```
This can also be changed to the single line constructor invocation.

cketti reviewed (changes requested):
Almost there :+1:

Note: After changing a pull request, it's usually a good idea to do a self review to see if you missed something (like applying a change to all occurrences of a particular pattern).

bernhard-fp commented:
@cketti Yes, I must have lost a bit of concentration there :-)
Did the changes you mentioned.

cketti reviewed (approved):
Thanks :+1:

In the future please don't update a pull request branch by merging `main` (or any other branch) into it. That makes it unnecessarily hard to clean up the history before merging. 
(This is also mentioned in the [pull request template](https://github.com/thunderbird/thunderbird-android/blob/main/.github/pull_request_template.md).)
